src/modules/item/NavigationIsland.tsx

```diff
@@ -0,0 +1,216 @@
+import { useNavigate, useParams } from 'react-router-dom';
+
+import ArrowBackIcon from '@mui/icons-material/ArrowBack';
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
+import ChatIcon from '@mui/icons-material/Forum';
+import ChatClosedIcon from '@mui/icons-material/ForumOutlined';
+import PinIcon from '@mui/icons-material/PushPin';
+import OutlinedPinIcon from '@mui/icons-material/PushPinOutlined';
+import { Box, Button, IconButton, Stack } from '@mui/material';
+
+import {
+  ActionTriggers,
+  DiscriminatedItem,
+  ItemTagType,
+  ItemType,
+} from '@graasp/sdk';
+
+import isArray from 'lodash.isarray';
+
+import { usePlayerTranslation } from '@/config/i18n';
+import { buildContentPagePath } from '@/config/paths';
+import { hooks, mutations } from '@/config/queryClient';
+import {
+  ITEM_CHATBOX_BUTTON_ID,
+  ITEM_PINNED_BUTTON_ID,
+} from '@/config/selectors';
+import { useLayoutContext } from '@/contexts/LayoutContext';
+import { PLAYER } from '@/langs/constants';
+
+const ChatButton = (): JSX.Element | null => {
+  const { t } = usePlayerTranslation();
+  const { itemId } = useParams();
+  const { data: item } = hooks.useItem(itemId);
+  const { toggleChatbox, isChatboxOpen } = useLayoutContext();
+
+  // do not show chatbox button is chatbox setting is not enabled
+  if (!item?.settings?.showChatbox) {
+    return null;
+  }
+
+  return (
+    <IconButton
+      id={ITEM_CHATBOX_BUTTON_ID}
+      color="primary"
+      onClick={toggleChatbox}
+      aria-label={
+        isChatboxOpen
+          ? t(PLAYER.HIDE_CHAT_TOOLTIP)
+          : t(PLAYER.SHOW_CHAT_TOOLTIP)
+      }
+    >
+      {isChatboxOpen ? <ChatIcon /> : <ChatClosedIcon />}
+    </IconButton>
+  );
+};
```

Comment:
I would have a folder for the island, and have separate files for each component.